Makefile

```diff
@@ -0,0 +1,22 @@
+BENCH_ITER=10
+
+.PHONY: memtests
+memtests: src/verify.c src/tests.c
+	clang -g $? -o ./$@ -Wall -Werror -lcrypto -lefence -DEXTRA_DEBUG -DBENCH_ITER=$(BENCH_ITER)
+	valgrind --leak-check=full ./$@
+
+.PHONY: ctests
+ctests: src/verify.c src/tests.c
+	clang -g $? -o ./$@ -Wall -Werror -lcrypto -DEXTRA_DEBUG -DBENCH_ITER=100000
+	./$@
+
+.PHONY: shell-tests
+	./test-cmdline.sh
+
+.PHONY: format
+format:
+	clang-format -i src/*.c src/*.h
+
+
+.PHONY: test
```

Comment:
This is a stylistic preference.  I put them together because if I remove the phony-target, the diff is localised to the area of the file where the rule is.

Reply:
Fair enough.  It works in my testing, too.